auth0/resource_auth0_guardian.go

```diff
@@ -118,24 +128,46 @@ func updateGuardian(d *schema.ResourceData, m interface{}) (err error) {
 			err = api.Guardian.MultiFactor.UpdatePolicy(&management.MultiFactorPolicies{p})
 		}
 	}
-	//TODO: Extend for other MFA types
+	if err := updatePhoneFactor(d, api); err != nil {
+		return err
+	}
+	if err := updateEmailFactor(d, api); err != nil {
+		return err
+	}
+	return readGuardian(d, m)
+}
+
+func updatePhoneFactor(d *schema.ResourceData, api *management.Management) error {
```

Comment:
We could probably simplify the logic inside this func as follows:

```suggestion
func updatePhoneFactor(d *schema.ResourceData, api *management.Management) error {
    ok, err := factorShouldBeUpdated(d, "phone")
    if err != nil {
        return err
    }
    if ok {
        if err := configurePhone(d, api); err != nil {
            return err
        }
    }
    
    return api.Guardian.MultiFactor.Phone.Enable(ok)
}
```

Reply:
That makes sense, thanks @phil-f ! Awesome work here